docs/how-tos/create-nexus-files-by-python-and-validate-them.md

```diff
@@ -0,0 +1,940 @@
+
+The second example was for the "software\_TYPE" attribute @URL entry in the "NXoptical\_spectroscopy" definition. Here the problem was that "url" was used instead of "URL".
+
+
+
+
+
+
+
+# 3\.4.1 Installation
+
+This is installed with pip:
+
+```
+pip install pynxtools
+```
+
+
+
+
+
+# 3\.4.2 Using the read\_nexus function
+
+Open your terminal. Assuming there is a folder at:
+
+For Linux:
+
+```
+/home/USER/nexusvalidation
+```
+
+For Windows:
+
+```
+C:\nexusvalidation
+```
+
+Put into this folder your NeXus file, for example the Raman.nxs file (INSERT LINK).
+
+Then use:
+
+```
+read_nexus -f C:\nexusvalidation\Raman.nxs
+```
+
+shows the output like this:
+
+```
+===== FIELD (//entry/data/spectrum_data_y): <HDF5 dataset "spectrum_data_y": shape (1600,), type "<f8">
+DEBUG: ===== FIELD (//entry/data/spectrum_data_y): <HDF5 dataset "spectrum_data_y": shape (1600,), type "<f8">
+value: [ 288.5499878  289.         288.4500122 ... 1875.        1889.349976 ...
+DEBUG: value: [ 288.5499878  289.         288.4500122 ... 1875.        1889.349976 ...
+Dataset referenced as NXdata SIGNAL
+DEBUG: Dataset referenced as NXdata SIGNAL
+===== ATTRS (//entry/data/spectrum_data_y@long_name)
+DEBUG: ===== ATTRS (//entry/data/spectrum_data_y@long_name)
+value: Raman Intensity 
+DEBUG: value: Raman Intensity 
+Dataset referenced as NXdata SIGNAL
+DEBUG: Dataset referenced as NXdata SIGNAL
+===== ATTRS (//entry/data/spectrum_data_y@units)
+DEBUG: ===== ATTRS (//entry/data/spectrum_data_y@units)
+value: counts 
+DEBUG: value: counts 
+
+DEBUG: 
+For Axis #0, 1 axes have been identified: [<HDF5 dataset "spectrum_data_x_Raman": shape (1600,), type "<f8">]
+DEBUG: For Axis #0, 1 axes have been identified: [<HDF5 dataset "spectrum_data_x_Raman": shape (1600,), type "<f8">]
+```
+
+Search for filed which are not found in the NeXus definiton by searching for the line: "DEBUG: NOT IN SCHEMA". Recheck the used NeXus definition to eliminate the problem. Be careful with upper and lower case notation and correct spelling.
+
+Keep in mind, that the output provides quite some information. This is useful for software development, but may be a bit too much for validation purposes.
+
+Similar features as the tables and messages provided from punx and cnxvalidate for pynxtools are planned to be implemented in the future ([https://github.com/FAIRmat-NFDI/pynxtools/pull/333](<https://github.com/FAIRmat-NFDI/pynxtools/pull/333>)).
+
+# 4\. Summary
+
+This tutorial showed:
+
+1. How to create a NeXus file with python.
+
+2. How to check if the NeXus file is valid.
+
+This provides the basics and fundamentals to create FAIR data, based on NeXus definitions. If your experimental setup provides enough meta data, you can extend the NeXus file creation script, to automaticalally include this information (e.g. measured spectra, sensor temperature, stage positon).
+
+### Pynxtools Parsers:
```

Comment:
There is a page at https://fairmat-nfdi.github.io/pynxtools/reference/plugins.html with all plugins. In order to keep single source principle, you can just link there.